sklearn/tree/_tree.pyx

```diff
@@ -223,7 +226,9 @@ cdef class DepthFirstTreeBuilder(TreeBuilder):
                     impurity = splitter.node_impurity()
                     first = 0
 
-                is_leaf = is_leaf or (impurity <= MIN_IMPURITY_SPLIT)
+                is_leaf = (is_leaf or
+                           (impurity <= MIN_IMPURITY_SPLIT) or
+                           (impurity < min_impurity_split))
```

Comment:
i agree! I renamed it to `LEAF_MIN_IMPURITY` but i think that's also a little bit confusing. do you have any suggestions for suitable names?